src/components/Header/Header.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { USER_ID, addTodos, updateTodo } from '../../api/todos';
+import { ErrorMessages } from '../../types/ErrorMessages';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todos: Todo[];
+  setTodos: (todos: Todo[]) => void;
+  setTempTodo: (tempTodo: Todo | null) => void;
+  setErrorMessage: (message: string) => void;
+  loadingIds: number[];
+  setLoadingIds: React.Dispatch<React.SetStateAction<number[]>>;
+}
+
+export const Header: React.FC<Props> = ({
+  todos,
+  setTodos,
+  setTempTodo,
+  setErrorMessage,
+  loadingIds,
+  setLoadingIds,
+}) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [title, setTitle] = useState('');
+  const [loading, setLoading] = useState(false);
+
+  const allCompleted = todos.every(todo => todo.completed);
+
+  useEffect(() => {
+    inputRef.current?.focus();
+  }, [todos, loading]);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const trimmedTitle = title.trim();
+
+    if (!trimmedTitle) {
+      setErrorMessage(ErrorMessages.EmptyTitle);
+
+      return;
+    }
+
+    setLoading(true);
+
+    const newTodo: Todo = {
+      id: 0,
+      userId: USER_ID,
+      title: trimmedTitle,
+      completed: false,
+    };
+
+    setTempTodo({ ...newTodo, id: 0 });
```

Comment:
```suggestion
    setTempTodo(newTodo);
```
The id is the same as in `newTodo`